mongodbatlas/resource_mongodbatlas_org_invitation.go

```diff
@@ -0,0 +1,272 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func resourceMongoDBAtlasOrgInvitation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasOrgInvitationCreate,
+		ReadContext:   resourceMongoDBAtlasOrgInvitationRead,
+		DeleteContext: resourceMongoDBAtlasOrgInvitationDelete,
+		UpdateContext: resourceMongoDBAtlasOrgInvitationUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasOrgInvitationImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"org_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"invitation_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"roles": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
```

Comment:
You can add a validation or valid values instead of `validateOrgRoles`. Here the example of use

```
Type: schema.TypeString,
ValidateFunc: validation.StringInSlice([]string{
"ORG_OWNER",
"ORG_GROUP_CREATOR",
"ORG_BILLING_ADMIN",
"ORG_READ_ONLY",
"ORG_MEMBER",
}, false),
```

Reply:
Done